src/components/SessionEntry.vue

```diff
@@ -31,119 +31,148 @@
 import groupAPIService from "@/services/API/groupData";
-import useAssets from '@/assets/assets.js'
+import useAssets from "@/assets/assets.js";
 
 const assets = useAssets();
-const EXTRA_5_MINUTES = 900000
+const EXTRA_5_MINUTES = 900000;
 export default {
   name: "SessionEntry",
-  components:{
+  components: {
     NoClassMessage,
-    Entry
+    Entry,
   },
   props: {
     sessionData: {
       type: Object,
       default() {
-        return {}
-      }
+        return {};
+      },
     },
   },
-  data(){
+  data() {
     return {
-      sessionActive : true, // whether the timings between the session and user match
+      sessionActive: true, // whether the timings between the session and user match
       groupData: null, //stores details about the group
-      currentDateTime : new Date(), // user's current date and time
+      currentDateTime: new Date(), // user's current date and time
       isLoading: false,
       loadingSpinnerSvg: assets.loadingSpinnerSvg,
-    }
+    };
   },
-  computed:{
+  computed: {
     /** Returns formatted local date of the user  */
-    currentDate(){
-      let year = this.currentDateTime.getFullYear().toString()
-      let month = (((this.currentDateTime.getMonth() + 1) < 9) ? '0' : '') + (this.currentDateTime.getMonth() + 1).toString()
-      let date = (((this.currentDateTime.getDate() + 1) < 9) ? '0' : '') + (this.currentDateTime.getDate()).toString()
-      return (year + '-' + month + '-' + date)
+    currentDate() {
+      let year = this.currentDateTime.getFullYear().toString();
+      let month =
+        (this.currentDateTime.getMonth() + 1 < 9 ? "0" : "") +
+        (this.currentDateTime.getMonth() + 1).toString();
+      let date =
+        (this.currentDateTime.getDate() + 1 < 9 ? "0" : "") +
+        this.currentDateTime.getDate().toString();
+      return year + "-" + month + "-" + date;
     },
 
-    /** Combines session start date and start time */
-    sessionStartDateTime(){
-      return (this.sessionData.startDate <= this.currentDate ) && (new Date(this.currentDate + " " + this.sessionData.startTime))
+    /** Checks if current date is greater than start date and combines current date and start time */
+    sessionStartDateTime() {
+      return (
+        this.sessionData.startDate <= this.currentDate &&
+        new Date(this.currentDate + " " + this.sessionData.startTime)
+      );
     },
 
-    /** Combines session end date and end time */
-    sessionEndDateTime(){
-      return (new Date(this.currentDate + " " + this.sessionData.endTime))
+    /** Checks if current date is less than end date and combines current date and end time */
+    sessionEndDateTime() {
+      if ("endDate" in this.sessionData) {
+        return (
+          this.sessionData.endDate >= this.currentDate &&
+          new Date(this.currentDate + " " + this.sessionData.endTime)
+        );
+      }
+      return new Date(this.currentDate + " " + this.sessionData.endTime);
```

Comment:
I didn't understand the purpose of putting an if check and the `this.sessionData.endDate >= this.currentDate &&` check because in both cases we are returning `return new Date(this.currentDate + " " + this.sessionData.endTime);`

Reply:
So the latter return is for sessions that don't have an endDate. In that case, we just build the datetime string and compare. The former is for when endDate is given. So it would return false if the current date has crossed the end date and will throw a no class message.